Filter/sorting

angho8 opened the pull request:
- New ListingComponent for displaying and filtering product listings.
- Search by title, filter by category, price range, and condition (New/Used).
- Sort by price (ascending/descending), newest/oldest, and distance.
- Filters and sorting are applied dynamically based on user inputs.

Closes #91

angho8 committed:
Create filter-sorting.js

angho8 committed:
filtering and sorting

zhandavidz commented on package.json:
```diff
@@ -34,5 +34,9 @@
+  "dependencies": {
+    "@types/react": "^19.0.10",
+    "react": "^19.0.0"
   }
```
`react` and `@types/react` are already added as dependencies in `/frontend/react`, so they shouldn't be needed to be added as dependencies in the root `package.json` as well

zhandavidz replied:
I think this still needs to be fixed in the updated version

zhandavidz commented on frontend/src/tsconfig.json:
```diff
@@ -0,0 +1,9 @@
+{
```
You shouldn't need a `tsconfig.json` in this folder as one is already defined in `/frontend`, lmk if you are having any issues running TypeScript.

zhandavidz commented on frontend/src/declarations.d.ts:
```diff
@@ -0,0 +1,4 @@
+declare module "*.json" {
```
I'm not quite sure what this file is doing, but it probably shouldn't be in `/frontend/src/`, we should discuss this further :)

zhandavidz commented on frontend/src/data.json:
```diff
@@ -0,0 +1,26 @@
+[
```
While it's not ideal to have this data sitting in `src/`, it's on me that the mock data api isn't up yet, so for now LGTM

zhandavidz commented on frontend/src/components/listing-card/filter-sorting.css:
```diff
@@ -0,0 +1,169 @@
+.listing-container {
```
Try to use TailwindCSS instead of `.css` files, we can talk discuss this further tmr :)

zhandavidz replied:
It looks good now!

zhandavidz reviewed (changes requested):
Overall looks good, it's failing a few prettier (which isn't super important for now) and there are a few config changes that are trying to be made that we should figure out, talk more tmr and we should be good to merge!

lhlee-umass reviewed (commented):
Looks very clean Angela!!
I'm sure we'll discuss today-- I'm also not quite sure what the .json files do but they were probably automatically added as you coded. 
Hopefully once I integrate you and Landy's PRs it might look right on the front end.

angho8 committed:
fixes/updates

zhandavidz commented on frontend/tsconfig.json:
```diff
@@ -3,5 +3,13 @@
+    "lib": ["es2016", "dom"]
 }
+}
+
```
I'm not sure these additions are necessary? We should discuss :)

zhandavidz reviewed (changes requested):
Love the new use of TailwindCSS!! Looking much better, there's a few more minor things with config files but should be good to merge soon

lhlee-umass reviewed (commented):
Looks great to me thanks Angela!

zhandavidz committed:
fix: Fix some config file issues

- updated tsconfig.json to be more uniform with rest of configs
- updated package.json to use existing dependencies

Resolves PR #95

zhandavidz committed:
misc: Format the code with ESLint and Prettier

- ran `npm run format` to automatically fix ESLint and Prettier issues

Resolves PR #95
